src/library/scala/sys/process/ProcessImpl.scala

```diff
@@ -32,14 +35,14 @@ private[process] trait ProcessImpl {
   }
   private[process] object Future {
     def apply[T](f: => T): (Thread, () => T) = {
-      val result = new SyncVar[Either[Throwable, T]]
+      val result = Promise[Either[Throwable, T]]()
       def run(): Unit =
-        try result.put(Right(f))
-        catch { case e: Exception => result.put(Left(e)) }
+        try result.trySuccess(Right(f))
+        catch { case e: Exception => result.trySuccess(Left(e)) }
 
       val t = Spawn(run())
 
-      (t, () => result.get match {
+      (t, () => Await.result(result.future,Duration.Inf) match {
```

Comment:
I'm not a huge fan of infinite blocking.

Reply:
So that will need a rewrite. The original implementation will block forever too,I have no idea how long a timeout fit.

Reply:
True. At least the new code will let the threadpool take that blocking into consideration since Await uses BlockContext.

Reply:
I want to first to keep the `scala.concurrent` small and elegant, then try to see if people want to move the `scala.io` and `scala.sys.process` to separate modules. So I just do simple replacement here :( not a rewrite.

Reply:
Understood, and agreed. :)